src/status_im/contexts/communities/events.cljs

```diff
@@ -184,3 +184,7 @@
                      :params     []
                      :on-success #(rf/dispatch [:communities/fetched-collapsed-categories-success %])
                      :on-error   #(log/error "failed to fetch collapsed community categories" %)}]}))
+
+(rf/reg-event-fx :communities/set-addresses-for-permissions
+ (fn [{:keys [db]} [addresses]]
+   {:db (assoc-in db [:communities/addresses-for-permissions] addresses)}))
```

Comment:
I am not sure if this was discussed in prev meetings. From what I understand, each community typically manages its own :communities/addresses-for-permissions. 

I was thinking, might it be more streamlined to tie the community ID directly to the permissions? We could do something like this:

(assoc-in db [:communities/addresses-for-permissions community-id] addresses)

Or assoc this into existing community?

Reply:
We haven't discussed this yet. Currently, I'm using `:communities/addresses-for-permissions` solely to store the selected addresses for the community we're attempting to join. Each time we open the request-to-join page, it updates the key with all addresses, allowing users to modify their selection. At the moment, I'm only utilizing it for the request-to-join UI.

Your suggestion to tie or link it with community data makes sense. However, I'm uncertain if we'll receive shared addresses from status-go once we join a community. If that's possible, keeping that information in the database tied to the community would be more reliable. I'll check if status-go shares this information. Please let me know if you have any insights on this.

Feel free to correct me if my thought process seems incorrect.

Reply:
Checked the UI again. I'm fine with this solution.